File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        hashmap = {}
        
        for idx,ele in enumerate(nums):
            
            if target - ele in hashmap:
                
                return [hashmap[target-ele],idx]
            
            hashmap[ele] = idx
        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
     
        int res[] = new int[2];
        
        Map<Integer,Integer> mapp = new HashMap<>();
        
        for(int i=0;i<nums.length;++i){
            
            int req = target - nums[i];
            
            if(mapp.containsKey(req)) 
            {
                return new int[] {mapp.get(req),i};
            }
            mapp.put(nums[i],i);
            
        }
    
        return null;
        
           
    }
}